### ISSUE TITLE
ban IP could still visit

### ISSUE THREAD

**Issue by zhanwen-cheng** [fail2ban/fail2ban#3015]
Hi, I have set findtime and maxretry in my  ```jail.local```  as below:
```
# "bantime" is the number of seconds that a host is banned.
bantime  = 10000000m
# A host is banned if it has generated "maxretry" during the last "findtime"
# seconds.
findtime  = 60
# "maxretry" is the number of failures before a host get banned.
maxretry = 5
```
and it show it really ban bad IP visitting with ```fail2ban-client status sshd``` command:
```
Status for the jail: sshd
|- Filter
|  |- Currently failed: 0
|  |- Total failed:     2606
|  `- File list:        /var/log/auth.log
`- Actions
   |- Currently banned: 14
   |- Total banned:     14
   `- Banned IP list:   (banned IP not showed here, only last one) 46.101.198.81                                                                              
```

However, when I run ```lastb -i ```, it showed ```46.101.198.81``` was banned after a lot of tried in more than 60 seconds:
```
ly       ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
ly       ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
ly       ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
ly       ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
ly       ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
ly       ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
ly       ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
ly       ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
wj       ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
wj       ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
wj       ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
wj       ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
wj       ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
wj       ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
wj       ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
wj       ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
peng     ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
pegn     ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
peng     ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
pegn     ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
peng     ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
peng     ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
peng     ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
peng     ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
arma3    ssh:notty    46.101.198.81    Sat May  1 05:28 - 05:28  (00:00)
arma3    ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
arma3    ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
arma3    ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
arma3    ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
arma3    ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
arma3    ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
arma3    ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
icter    ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
icter    ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
ectrip   ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
ectrip   ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
plane    ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
plane    ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
plane    ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
plane    ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
plane    ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
plane    ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
plane    ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
plane    ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
air      ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
air      ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
air      ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
air      ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
air      ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
air      ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
air      ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
air      ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
lava     ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
lava     ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
lava     ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
lava     ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
lava     ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
lava     ssh:notty    46.101.198.81    Sat May  1 05:27 - 05:27  (00:00)
...
root     ssh:notty    46.101.198.81    Sat May  1 04:52 - 04:52  (00:00)
root     ssh:notty    46.101.198.81    Sat May  1 04:52 - 04:52  (00:00)
root     ssh:notty    46.101.198.81    Sat May  1 04:52 - 04:52  (00:00)
root     ssh:notty    46.101.198.81    Sat May  1 04:52 - 04:52  (00:00)
root     ssh:notty    46.101.198.81    Sat May  1 04:52 - 04:52  (00:00)
root     ssh:notty    46.101.198.81    Sat May  1 04:52 - 04:52  (00:00)
```

Is this normal? How to limit 46.101.198.81 visit as soon as it been banned?

**Comment by sebres** [comment]
I guess if you'll take a look in fail2ban.log, you would see a lot of `Ban` and `Already banned` messages for this IP.
If so your banning action seems not to work properly. 

Check whether you'd see some errors in log after the ban.
Also check your net-filter subsystem (firewalld? ufw?) e. g. default or own whitelisting rules etc.
Also note https://github.com/fail2ban/fail2ban/issues/2545#issuecomment-543087971 and below.

**Comment by zhanwen-cheng** [comment]
Hi sebres, thanks for you replying. I check the ```/var/log/fail2ban.log``` file, there is no erros.
<details>
<summary>fail2ban.log</summary>
<pre><code>
2021-05-04 18:19:58,541 fail2ban.server         [45252]: INFO    --------------------------------------------------
2021-05-04 18:19:58,542 fail2ban.server         [45252]: INFO    Starting Fail2ban v0.10.2
2021-05-04 18:19:58,545 fail2ban.database       [45252]: INFO    Connected to fail2ban persistent database '/var/lib/fail2ban/fail2ban.sqlite3'
2021-05-04 18:19:58,548 fail2ban.database       [45252]: WARNING New database created. Version '2'
2021-05-04 18:19:58,549 fail2ban.jail           [45252]: INFO    Creating new jail 'sshd'
2021-05-04 18:19:58,578 fail2ban.jail           [45252]: INFO    Jail 'sshd' uses pyinotify {}
2021-05-04 18:19:58,585 fail2ban.jail           [45252]: INFO    Initiated 'pyinotify' backend
2021-05-04 18:19:58,586 fail2ban.filter         [45252]: INFO      maxLines: 1
2021-05-04 18:19:58,610 fail2ban.server         [45252]: INFO    Jail sshd is not a JournalFilter instance
2021-05-04 18:19:58,610 fail2ban.filter         [45252]: INFO    Added logfile: '/var/log/auth.log' (pos = 0, hash = a62eb08e50490c9d69790bfb692cd38aec9341b5)
2021-05-04 18:19:58,614 fail2ban.filter         [45252]: INFO      encoding: UTF-8
2021-05-04 18:19:58,614 fail2ban.filter         [45252]: INFO      maxRetry: 5
2021-05-04 18:19:58,614 fail2ban.filter         [45252]: INFO      findtime: 60
2021-05-04 18:19:58,615 fail2ban.actions        [45252]: INFO      banTime: 600000000
2021-05-04 18:19:58,617 fail2ban.jail           [45252]: INFO    Jail 'sshd' started
2021-05-04 18:22:02,580 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:22:02
2021-05-04 18:22:04,645 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:22:04
2021-05-04 18:22:05,013 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:22:05
2021-05-04 18:22:06,749 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:22:06
2021-05-04 18:22:38,179 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:22:38
2021-05-04 18:22:38,819 fail2ban.actions        [45252]: NOTICE  [sshd] Ban 10.16.14.202
2021-05-04 18:22:38,884 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:22:38
2021-05-04 18:22:39,117 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:22:39
2021-05-04 18:22:40,309 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:22:40
2021-05-04 18:22:46,548 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:22:46
2021-05-04 18:22:46,821 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:22:46
2021-05-04 18:22:46,861 fail2ban.actions        [45252]: NOTICE  [sshd] 10.16.14.202 already banned
2021-05-04 18:22:47,077 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:22:47
2021-05-04 18:22:48,222 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:22:48
2021-05-04 18:22:55,332 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:22:55
2021-05-04 18:22:55,421 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:22:55
2021-05-04 18:22:55,589 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:22:55
2021-05-04 18:22:56,073 fail2ban.actions        [45252]: NOTICE  [sshd] 10.16.14.202 already banned
2021-05-04 18:22:56,397 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:22:56
2021-05-04 18:23:02,306 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:23:02
2021-05-04 18:23:02,541 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:23:02
2021-05-04 18:23:02,765 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:23:02
2021-05-04 18:23:03,525 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:23:03
2021-05-04 18:23:04,084 fail2ban.actions        [45252]: NOTICE  [sshd] 10.16.14.202 already banned
2021-05-04 18:23:44,629 fail2ban.filter         [45252]: INFO    [sshd] Found 10.16.14.202 - 2021-05-04 18:23:44
2021-05-04 18:33:51,271 fail2ban.server         [45252]: INFO    Shutdown in progress...
2021-05-04 18:33:51,271 fail2ban.server         [45252]: INFO    Stopping all jails
2021-05-04 18:33:51,272 fail2ban.filter         [45252]: INFO    Removed logfile: '/var/log/auth.log'
2021-05-04 18:33:51,709 fail2ban.actions        [45252]: NOTICE  [sshd] Flush ticket(s) with iptables-multiport
2021-05-04 18:33:51,718 fail2ban.actions        [45252]: NOTICE  [sshd] Unban 10.16.14.202
2021-05-04 18:33:51,733 fail2ban.jail           [45252]: INFO    Jail 'sshd' stopped
2021-05-04 18:33:51,734 fail2ban.database       [45252]: INFO    Connection to database closed.
2021-05-04 18:33:51,734 fail2ban.server         [45252]: INFO    Exiting Fail2ban
2021-05-04 18:33:51,889 fail2ban.server         [46413]: INFO    --------------------------------------------------
2021-05-04 18:33:51,889 fail2ban.server         [46413]: INFO    Starting Fail2ban v0.10.2
2021-05-04 18:33:51,892 fail2ban.database       [46413]: INFO    Connected to fail2ban persistent database '/var/lib/fail2ban/fail2ban.sqlite3'
2021-05-04 18:33:51,894 fail2ban.jail           [46413]: INFO    Creating new jail 'sshd'
2021-05-04 18:33:51,913 fail2ban.jail           [46413]: INFO    Jail 'sshd' uses pyinotify {}
2021-05-04 18:33:51,924 fail2ban.jail           [46413]: INFO    Initiated 'pyinotify' backend
2021-05-04 18:33:51,926 fail2ban.filter         [46413]: INFO      maxLines: 1
2021-05-04 18:33:51,952 fail2ban.server         [46413]: INFO    Jail sshd is not a JournalFilter instance
2021-05-04 18:33:51,953 fail2ban.filter         [46413]: INFO    Added logfile: '/var/log/auth.log' (pos = 1486799, hash = a62eb08e50490c9d69790bfb692cd38aec9341b5)
2021-05-04 18:33:51,956 fail2ban.filter         [46413]: INFO      encoding: UTF-8
2021-05-04 19:17:42,522 fail2ban.filter         [48560]: INFO      maxRetry: 5
2021-05-04 19:17:42,522 fail2ban.filter         [48560]: INFO      findtime: 60
2021-05-04 19:17:42,523 fail2ban.actions        [48560]: INFO      banTime: 600000000
2021-05-04 19:17:43,743 fail2ban.jail           [48560]: INFO    Jail 'sshd' started
2021-05-04 19:17:44,528 fail2ban.actions        [48560]: NOTICE  [sshd] Restore Ban 10.16.14.202
2021-05-04 19:19:45,241 fail2ban.actions        [48560]: NOTICE  [sshd] Unban 10.16.14.202
2021-05-04 19:21:58,244 fail2ban.filter         [48560]: INFO    [sshd] Found 101.200.41.63 - 2021-05-04 19:21:57
2021-05-04 19:22:00,296 fail2ban.filter         [48560]: INFO    [sshd] Found 101.200.41.63 - 2021-05-04 19:22:00
2021-05-04 19:22:03,001 fail2ban.filter         [48560]: INFO    [sshd] Found 101.200.41.63 - 2021-05-04 19:22:02
2021-05-04 19:22:09,412 fail2ban.filter         [48560]: INFO    [sshd] Found 101.200.41.63 - 2021-05-04 19:22:09
2021-05-04 19:22:11,176 fail2ban.filter         [48560]: INFO    [sshd] Found 101.200.41.63 - 2021-05-04 19:22:11
2021-05-04 19:22:11,496 fail2ban.actions        [48560]: NOTICE  [sshd] Ban 101.200.41.63
2021-05-04 19:22:13,165 fail2ban.filter         [48560]: INFO    [sshd] Found 101.200.41.63 - 2021-05-04 19:22:13
2021-05-04 19:22:25,892 fail2ban.filter         [48560]: INFO    [sshd] Found 101.200.41.63 - 2021-05-04 19:22:25
2021-05-04 19:22:28,598 fail2ban.filter         [48560]: INFO    [sshd] Found 101.200.41.63 - 2021-05-04 19:22:28
2021-05-04 19:22:30,201 fail2ban.filter         [48560]: INFO    [sshd] Found 101.200.41.63 - 2021-05-04 19:22:30
2021-05-04 19:22:34,509 fail2ban.filter         [48560]: INFO    [sshd] Found 101.200.41.63 - 2021-05-04 19:22:34
2021-05-04 19:22:34,743 fail2ban.actions        [48560]: NOTICE  [sshd] 101.200.41.63 already banned
2021-05-04 19:22:38,148 fail2ban.filter         [48560]: INFO    [sshd] Found 101.200.41.63 - 2021-05-04 19:22:38
2021-05-04 19:30:35,908 fail2ban.filter         [48560]: INFO    [sshd] Found 101.200.41.63 - 2021-05-04 19:30:34
2021-05-04 19:35:21,058 fail2ban.filter         [48560]: INFO    [sshd] Found 101.200.41.63 - 2021-05-04 19:35:21
</code></pre>
</details>
<details>
<summary>iptables -nL</summary>
<pre><code>
root@profxia-SUSTC-309:~# iptables -nL
Chain INPUT (policy ACCEPT)
target     prot opt source               destination
f2b-sshd   tcp  --  0.0.0.0/0            0.0.0.0/0            multiport dports 22
DROP       all  --  68.183.95.28         0.0.0.0/0

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination

Chain f2b-sshd (1 references)
target     prot opt source               destination
REJECT     all  --  101.200.41.63        0.0.0.0/0            reject-with icmp-port-unreachable
RETURN     all  --  0.0.0.0/0            0.0.0.0/0

</code></pre>
</details>

As I am testing ```101.200.41.63``` to visit my server, I noticed that there is ```f2b-sshd``` chian in iptables rules(i don't have ufw or other firewall), it seemed that this ```101.200.41.63``` is rejected in ```f2b-sshd``` chian, but I can still login in via this IP.

Also, if I block IP manually with ```iptables -I INPUT -s 101.200.41.63 -j DROP```, the ```iptables -nL``` look like below(the blocked IP added to chain INPUT rather than f2b-sshd),.And in this way, I could not login via ```101.200.41.63```.
<details>
<summary>iptables -nL</summary>
<pre><code>
root@profxia-SUSTC-309:~# iptables -nL
Chain INPUT (policy ACCEPT)
target     prot opt source               destination
DROP       all  --  101.200.41.63        0.0.0.0/0
f2b-sshd   tcp  --  0.0.0.0/0            0.0.0.0/0            multiport dports 22
DROP       all  --  68.183.95.28         0.0.0.0/0

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination

Chain f2b-sshd (1 references)
target     prot opt source               destination
REJECT     all  --  101.200.41.63        0.0.0.0/0            reject-with icmp-port-unreachable
RETURN     all  --  0.0.0.0/0            0.0.0.0/0
</code></pre>
</details>

Does that mean the ```f2b``` rule I got here is wrong?
Actually I didn't edit any rule to iptables.How to solve that?

Sorry, I am really not good at those firewall issues. Sorry to disrupt you.

**Comment by sebres** [comment]
> As I am testing ... to visit my server

Is your sshd port 22? Jail `sshd` uses multi-port banning action per default. If you had changed your sshd-port (or have still another listener on other ports), you must change `port` settings for sshd jail.

> Does that mean the `f2b` rule I got here is wrong?

No, it looks correct to me, but as already said it works only for port 22 at the moment (so net-filter rejecting incoming packets on destination port 22).
If you nevertheless need similar to your DROP handling for this jail, you can simply set following in your jail.local:
```ini
[sshd]
banaction = iptables-allports[blocktype=DROP]
```
(this will use `iptables-allports`, so banning occurs for all ports and it will use drop instead of reject)

But I don't see why reject shouldn't work here. This is simply different behavior how the port will look for the other side (see https://github.com/fail2ban/fail2ban/issues/2217#issuecomment-419881481 for details), but the packets (so incoming connections) are normally not possible anymore for this IP, no matter whether they are getting properly rejected or dropped.

**Comment by zhanwen-cheng** [comment]
Hi, sebres, thanks for your help.

> Is your sshd port 22?

No, we have opened port 22 and 11022 for ssh.

Here, I added port 22 and 11022 into my [sshd],and set banaction into ```iptables-allports[blocktype=DROP]```, 
**And it WORKS!!!**

Although I am still not very faimlar with the iptables setting and jail parameters, this issue is solved with your suggestions! **Thank you again!!!**
I hope this could help anyone who get the same issue.
